src/app.ts

```diff
@@ -80,9 +84,13 @@ function withPandaAuth(
 			if (panAuthResult.status === 'Authorised') {
 				onSuccess(panAuthResult);
 			} else {
-				res.status(403).send({
-					error: 'Not authorised by pan-domain login',
-				});
+				if (onFailure) {
+					onFailure();
+				} else {
+					res.status(403).send({
+						error: 'Not authorised by pan-domain login',
+					});
+				}
```

Comment:
We could definitely save this up for a future PR, but it feels more natural to me to either always do the same thing here, or always have the caller provide the action to take when auth doesn't succeed (so the suggestion is to not built the default json response behaviour into this, but instead pass it in, i.e. make `onFailure` a required param to `withPandaAuth`). What do you think?